nifi-nar-bundles/nifi-html-bundle/nifi-html-processors/src/main/java/org/apache/nifi/GetHTMLElement.java

```diff
@@ -0,0 +1,243 @@
+            final Document doc = parseHTMLDocumentFromFlowfile(flowFile, context, session);
+            final Elements eles = doc.select(context.getProperty(CSS_SELECTOR)
+                    .evaluateAttributeExpressions().getValue());
+            final String prependValue = context.getProperty(PREPEND_ELEMENT_VALUE)
+                    .evaluateAttributeExpressions(flowFile).getValue();
+            final String appendValue = context.getProperty(APPEND_ELEMENT_VALUE)
+                    .evaluateAttributeExpressions(flowFile).getValue();
+
+            if (eles == null || eles.size() == 0) {
+                //No element found
+                session.transfer(flowFile, REL_NOT_FOUND);
+            } else {
+                for (final Element ele : eles) {
+                    final FlowFile ff = session.create();
+
+                    switch (context.getProperty(DESTINATION).getValue()) {
+                        case DESTINATION_ATTRIBUTE:
+                            final FlowFile atFlowfile = session.putAttribute(ff, HTML_ELEMENT_ATTRIBUTE_NAME,
+                                    extractElementValue(
+                                            prependValue,
+                                            context.getProperty(OUTPUT_TYPE).getValue(),
+                                            appendValue,
+                                            ele,
+                                            context.getProperty(ATTRIBUTE_KEY).evaluateAttributeExpressions()
+                                                    .getValue()));
+                            session.getProvenanceReporter().create(atFlowfile);
+                            session.transfer(atFlowfile, REL_SUCCESS);
+                            break;
+                        case DESTINATION_CONTENT:
+                            final FlowFile conFlowfile = session.write(ff, new StreamCallback() {
+                                @Override
+                                public void process(InputStream inputStream, OutputStream outputStream) throws IOException {
+                                    try {
+                                        outputStream.write(extractElementValue(
+                                                prependValue,
+                                                context.getProperty(OUTPUT_TYPE).getValue(),
+                                                appendValue,
+                                                ele,
+                                                context.getProperty(ATTRIBUTE_KEY).evaluateAttributeExpressions()
+                                                        .getValue()).getBytes());
+                                    } catch (Exception ex) {
+                                        session.transfer(ff, REL_FAILURE);
+                                    }
+                                }
+                            });
+
+                            session.getProvenanceReporter().create(conFlowfile);
```

Comment:
Same point as above, should be a FORK, not CREATE, but it'll be taken care of by the framework.